Assets/Scripts/KingAI.cs

```diff
@@ -1,51 +1,200 @@
 using System.Collections;
 using System.Collections.Generic;
+using Unity.VisualScripting;
 using UnityEngine;
 using UnityEngine.AI;
 
 public class KingAI : MonoBehaviour
 {
-    public Transform player; 
+    public enum FSMStates
+    {
+        Wander,
+        Chase,
+        Dialogue
+    }
+
+    public FSMStates currentState;
+
+    GameObject[] wanderPoints;
+    Vector3 nextDestination;
+    float distanceToPlayer;
+
+    public float chaseDistance = 10f;
+    public float dialogueRange = 5f;
+
+    public Transform npcEyes;
+    public float fov = 200f;
+
+    int currentDestinationIndex = 0;
+
+    public GameObject player; 
     private NavMeshAgent agent;
     private Animator anim;
 
+
     void Start()
     {
         agent = GetComponent<NavMeshAgent>();
 
         anim = GetComponent<Animator>();
         anim.SetInteger("WeaponType_int", 0);
 
-        player = GameObject.FindGameObjectWithTag("Player").transform;
+        currentState = FSMStates.Wander;
 
-        agent.SetDestination(player.position);
-        agent.stoppingDistance = 5f;
+        player = GameObject.FindGameObjectWithTag("Player");
+        wanderPoints = GameObject.FindGameObjectsWithTag("WanderPoint");
+
+        agent.SetDestination(player.transform.position);
+        agent.stoppingDistance = 0f;
     }
 
     void Update()
     {
-        FacePlayer();
 
-        float distanceToPlayer = Vector3.Distance(player.position, transform.position); ;
+        distanceToPlayer = Vector3.Distance(player.transform.position, transform.position); ;
+
+        switch (currentState)
+        {
+            case FSMStates.Wander:
+                UpdateWanderState();
+                break;
+            case FSMStates.Chase:
+                UpdateChaseState();
+                break;
+            case FSMStates.Dialogue:
+                UpdateDialogueState();
+                break;
+        }
+        //if (distanceToPlayer > 5f)
+        //{
+        //    agent.speed = 3.5f;
+        //    agent.SetDestination(player.position);
+        //    anim.SetFloat("Speed_f", 0.7f);
+        //    anim.SetInteger("Animation_int", 0);
+        //} else
+        //{
+        //    anim.SetFloat("Speed_f", 0f);
+        //    anim.SetInteger("Animation_int", 1);
+        //}
+    }
+
+    void UpdateWanderState()
+    {
+        agent.speed = 3f;
+        anim.SetFloat("Speed_f", 0.5f);
```

Comment:
using 0.5 specifically as the speed is broken because theres a >0.5 condition to switch to running, and a <0.5 condition to switch back to walking, so do a value between 0.25 and 0.5